tests/utils.py

```diff
@@ -17,6 +17,9 @@ def generate_sample_records(
     run_type: str | None = "daily",
     action: str | None = "index",
     run_id: str | None = None,
+    year: str | int | None = "2024",
+    month: str | int | None = "12",
+    day: str | int | None = "1",
```

Comment:
Perhaps I'm missing something, but maybe we should omit these from the testing utility?  As we'd expect the post-init behavior to set these based on a passed `run_date`?

I've confirmed that tests still pass without them. 

If so, I think removing from the actual init a few lines down as well.

Reply:
If we do end up making these dynamic properties as proposed in [this comment](https://github.com/MITLibraries/timdex-dataset-api/pull/15#discussion_r1876795496), I think it naturally gets at why including these here feels a little off.

Reply:
These are now omitted as part of https://github.com/MITLibraries/timdex-dataset-api/pull/15/commits/5e532d365baf9b6f6db68d68dd09d20dabfece00.